linera-views/src/key_value_store_view.rs

```diff
@@ -605,8 +619,12 @@ where
             match operation {
                 WriteOperation::Delete { key } => {
                     ensure!(key.len() <= max_key_size, ViewError::KeyTooLong);
-                    if let Some(size) = self.sizes.get(&key).await? {
-                        self.total_size.sub_assign(size);
+                    if let Some(value) = self.sizes.get(&key).await? {
+                        let single_size = SizeData {
+                            key: key.len() as u32,
```

Comment:
Do we need to deal with the overflow?

Reply:
u32 corresponds to 4G.
So, it is a very big size, but not something that we cannot imagine.

Reply:
So it should probably be something like:

```suggestion
                            key: u32::try_from(key.len()).map_err(|_| ArithmeticError::Overflow)?,
```

Reply:
It surely doesn't hurt to use the safe APIs. We also have a notion of max key size that might prevent this (does it?)